### ISSUE TITLE
Move CQ Summary Data to Custom Message Type

### ISSUE THREAD

**Issue by bbernays** [cloudquery/cloudquery#17118]
Currently the CLI sends a single insert message and destination that need to add more data have to filter on the table name.

https://github.com/cloudquery/cloudquery/pull/17117

The cleaner and more scalable way of implementing this is to create a custom message type for this rather than having a specific if statement in the main write function

**Comment by yevgenypats** [comment]
do we really need a sync summary to be implemented in every destination though? or is it already there?

**Comment by bbernays** [comment]
Yes, every destination should be able to store the summary of the sync. We do not need to implement a custom message type if we are alright using the table name to add custom information (when it is necessary)

**Comment by erezrokah** [comment]
I marked this as low priority. We have it working in a single destination but the table name we use for sync summary is hard coded and coupled with the CLI's code. We can make it generic to all destinations via a designated protocol message if we need to add it to more destinations
